File: src/org/jetbrains/jet/j2k/visitors/Visitor.java
package org.jetbrains.jet.j2k.visitors;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.jet.j2k.ast.Element;

/**
 * @author ignatov
 */
public interface Visitor {
  @NotNull
  Element getResult();
}

File: compiler/testData/diagnostics/tests/platformTypes/methodCall/visitor.kt
// !WITH_NEW_INFERENCE
// FILE: p/Visitor.java

package p;

public interface Visitor<D> {

}

// FILE: p/Element.java

package p;

import org.jetbrains.annotations.NotNull;

public class Element {
    public <D, R> R accept(@NotNull Visitor<R> visitor, D data) {return null;}
}

// FILE: k.kt

import p.*

fun test(v: Visitor<Nothing>, e: Element) {
    e.accept(v, null)
}
